### PR TITLE
Adding Auth Token generation.

### PR DESCRIPTION
**Pull request by ananner**
Adding EdgeWorkers Auth Token generation that can be used in conjunction with debug headers.

### CODE REVIEW

**Comment by ananner** [comment]
cc @dctheobald @sscian

File: src/cli-main.ts [hunk 1]
```diff
@@ -253,6 +254,67 @@ program
...
+  exclusive to the --acl option; only use one or the other.")
+  .option("--expiry <expiry>", "The number of minutes during which the token is valid, after which it expires. Max value is 60 minutes; default value is 15 minutes.")
+  .action(async function (secretKey, options) {
+
+    if(!secretKey) {
+      cliUtils.logAndExit(1, "ERROR: The secret key specified in the property in which the EdgeWorker executes must be supplied in order to generate an authentication token.");
+    } else if(secretKey.length % 2 != 0) {
+      cliUtils.logAndExit(1, "ERROR: The secret key specified in the property in which the EdgeWorker executes must have an even number of hex characters.");
+    } else if(!secretKey.match(/^[0-9a-fA-F]+$/)) {
+      console.log("INVALID SECRET: " + secretKey);
```
Comment (dctheobald) [inline src/cli-main.ts:275]: Please remove the console.log as this is likely for debugging the new code here only.

File: src/cli-main.ts [hunk 2]
```diff
@@ -253,6 +254,67 @@ program
...
+      cliUtils.logAndExit(1, "ERROR: The secret key specified in the property in which the EdgeWorker executes must contain only hex characters: [0-9a-f]");
+    } else {
+      secretKey = secretKey.toLowerCase();
+    }
+
+    var expiry = 15; // Use 15 minutes as the default value
```
Comment (dctheobald) [inline src/cli-main.ts:281]: I dont think the if clause is necessary since you default to 15m anyway before it.

File: src/cli-main.ts [hunk 3]
```diff
@@ -701,5 +763,43 @@ async function createNewActivation(ewId: string, network: string, versionId: str
...
+
+  //console.log("Token: " + new_token);
+  //console.log("HMAC signing string: " + hmac_token);
+
+  const hexedSecretKey = CryptoJS.enc.Hex.parse(secretKey);
+  const hash = CryptoJS.HmacSHA256(hmac_token, hexedSecretKey);
+  const hashStr = CryptoJS.enc.Hex.stringify(hash);
+  var auth_token = new_token + field_delimiter + `hmac=${hashStr}`;
+
+  console.log("\nAdd the following request header to your requests to get additional trace information.\n");
```
Comment (dctheobald) [inline src/cli-main.ts:799]: Please use the appropriate log and exit function here so automation can still work here.  Console.log should be removed.

**Review by dctheobald — CHANGES_REQUESTED** [review]

File: src/cli-main.ts (hunk 1 above)
Comment (ananner) [reply src/cli-main.ts:275]: Removing that line.

File: src/cli-main.ts (hunk 3 above)
Comment (ananner) [reply src/cli-main.ts:799]: Changing this to use `cli.logWithBorder() `and `cli.logWithExit()`

File: src/cli-main.ts (hunk 2 above)
Comment (ananner) [reply src/cli-main.ts:281]: Fixing this.

**Comment by ananner** [comment]
@dctheobald, I think I'd need to also update the README file as well.

File: src/cli-main.ts [hunk 4]
```diff
@@ -253,6 +254,66 @@ program
...
+      } else if(expiry < 1 || expiry > 60) {
+        cliUtils.logAndExit(1, "ERROR: The expiry is invalid. It must be an integer value (in minutes) between 1 and 60.");
+      }
+    }
+
+    var path = "/*";
+    var isACL = true;
+    if(options.acl) {
+      if(options.url) {
+        cliUtils.logAndExit(1, "ERROR: The --acl and --url parameters are mutually exclusive; please use only one parameter. Specifying neither will result in a \
```
Comment (Ehsap) [inline src/cli-main.ts:296]: Formatting: When I hit this code path on my MacOS terminal this message is formatted with about 9 spaces between "a" and "default":
Screen Shot 2020-09-24 at 8 43 01 PM

Comment (ananner) [reply src/cli-main.ts:296]: Yes, I can see the same thing. I will fix and update.

**Pull request merged**